### PR TITLE
[SYCL] [FPGA] Disable swapping dimension values of WorkGroupAttr arguments for semantic checks

### PR DESCRIPTION
**Pull request by smanna12**
This is a followup comments on https://github.com/intel/llvm/pull/2906#discussion_r546940763
and https://github.com/intel/llvm/pull/2906#discussion_r548524994.

Currently for a SYCLDevice, first and third argument of WorkGroupAttr 
are reversed in doing semantic checks as wells as code generation.

There are few problems with this approach (swapping order in AST):
1. A user who writes the order as X, Y, Z in their source will get Z, Y, X when
   they pretty print the AST.
2. The same will happen when dumping the AST to text or JSON or
   generating diagnostic messages since the fist and third arguments of
   WorkGroupAttr are reversed.
3. This is very confusing for user to see different order than the original
   source codes for Semantic checks.

max_work_group_size() follows same rules as reqd_work_group_size() in Sema and LLVM IR generation.

This patch keeps the arguments in the same order as they were parsed in doing semantic checks,
and when performing code generation, we would change the argument order when lowering to
LLVM IR and not when creating the semantic attribute.

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

**Comment by smanna12** [comment]
could you please review @Fznamznon? Thanks

**Review by Fznamznon — DISMISSED** [review]
It would be great to hear from @AaronBallman here as well, but for me it looks fine.

**Comment by smanna12** [comment]
> It would be great to hear from @AaronBallman here as well, but for me it looks fine.

Thanks Mariya.

**Review by AaronBallman — DISMISSED** [review]
LGTM aside from a very minor nit that can be fixed in one of the tests.

File: clang/test/SemaSYCL/intel-max-global-work-dim-device.cpp [hunk 1]
```diff
@@ -94,13 +94,13 @@ int main() {
     h.single_task<class test_kernel5>(TRIFuncObjGood2());
     // CHECK-LABEL: FunctionDecl {{.*}}test_kernel5
     // CHECK:       ReqdWorkGroupSizeAttr {{.*}}
-    // // CHECK-NEXT:  IntegerLiteral{{.*}}1{{$}}
+    // // CHECK-NEXT:  IntegerLiteral{{.*}}4{{$}}
```
[the remark this answers is not in the document]
Comment (smanna12) [reply clang/test/SemaSYCL/intel-max-global-work-dim-device.cpp]: Done. Sorry for the duplicate comments.

**Review by Fznamznon — APPROVED** [review]

**Comment by smanna12** [comment]
Thank you everyone for the reviews.

**Review by elizabethandrews — APPROVED** [review]

**Review by MrSidims — APPROVED** [review]

**Pull request merged**